edgelet/doc/devguide.md

```diff
@@ -91,12 +91,14 @@ Clippy is a linting tool for rust. It provides suggestions for more idiomatic ru
 
 Install it with:
 ```
-rustup component add clippy-preview --toolchain=nightly
+rustup component add clippy-preview
```

Comment:
Per https://github.com/rust-lang/rust-clippy#step-2-install-clippy, shouldn't this be just "clippy" and not "clippy-preview"?

Reply:
Huh - I did "clippy-preview" last week and it worked - so I just tried"clippy" and it worked as well - will update.

Reply:
As of 1.32.0, 

> >rustup component add clippy
> error: toolchain 'stable-x86_64-pc-windows-msvc' does not contain component 'clippy' for target 'x86_64-pc-windows-msvc'
> 
> C:\Users\darobs>rustup component add clippy-preview
> info: downloading component 'clippy-preview'
> info: installing component 'clippy-preview'
> 

I'm going to write:
Install it with:
```
rustup component add clippy
```
if Clippy doesn't install (the toolchain does not contain component `clippy`), try:
```
rustup component add clippy-preview
```

Reply:
1. `clippy` is an alias for `clippy-preview`. The latter is still there for backward-compatibility.

1. Only updated rustup knows about `clippy`. If you get an error that `clippy` doesn't exist, then that's because you haven't run `rustup self update`.

So no, you shouldn't do the fallback. Instead update your rustup (and probably the CI as well).

Reply:
Also, since rustup caches the components available when you install a toolchain, updating rustup now will not make it aware of the `clippy` component, so it'll still think it doesn't exist. You'll also need to `remove` and re-`install` the stable toolchain in that case.

Reply:
Also, since rustup caches the components available when you install a toolchain, updating rustup now will not make it aware of the `clippy` component, so it'll still think it doesn't exist. You'll also need to `remove` and re-`install` the stable toolchain in that case.

Reply:
Thanks, @arsing, but `rustup self update` didn't fix windows (yet):

> C:\>rustup self update
> info: checking for self-updates
> 
> C:\>rustup component add clippy
> error: toolchain 'stable-x86_64-pc-windows-msvc' does not contain component 'clippy' for target 'x86_64-pc-windows-msvc'
> 
> C:\>rustup --version
> rustup 1.16.0 (beab5ac2b 2018-12-06)
>

Reply:
Hence my second comment.

Reply:
Uninstalling and reinstalling the toolchain is making me not like C
I removed the fallback, but added some information in the "additional tools" section that if you're running an older version of Rust, you should:
```
rustup self update
rustup uninstall stable
rustup install stable
```

Reply:
Yeah, it's just a one-time thing because of how the feature got implemented - component aliases were added to support this which requires updated rustup. So updating the toolchain with old rustup means the alias is never recognized until a toolchain is updated or reinstalled.